### DISCUSSION TITLE
Line jump

### DISCUSSION THREAD

**Discussion by jin-ahn** [folke/flash.nvim#22] [Ideas]
Could we add the ability to jump to a specific line on buffer? I know there's multiple ways in nvim to do it by default but implementing it here would save 2-3 key strokes

**Comment by max397574** [comment]
don't rly see how you could actually save keystrokes since it's just `<line nr>G` but anyways

```lua
require("flash").jump({
    search = { multi_window = true, wrap = true },
    highlight = { backdrop = true, label = { current = true } },
    matcher = function()
        local results = {}
        for i = vim.fn.line("w0"), vim.fn.line("w$") do
            table.insert(results, {
                pos = { i, 1 },
                end_pos = { i, 0 },
            })
        end
        return results
    end,
    action = function(match, _)
        vim.api.nvim_set_current_win(match.win)
        vim.api.nvim_win_call(match.win, function()
            vim.api.nvim_win_set_cursor(match.win, { match.pos[1], 0 })
        end)
    end,
})
```

**Reply by IndianBoy42** [reply]
This is my implementation that jumps vertically, so the cursor column is preserved
```lua
    matcher = function(win)
      local cur = vim.api.nvim_win_get_cursor(win)
      local wininfo = vim.api.nvim_win_call(win, function() return vim.fn.winsaveview() end)
      local top = wininfo.topline
      local h = vim.api.nvim_win_get_height(win)

      local res = {}
      for i = 1, h do
        local pos = { top + i, cur[2] }
        res[#res + 1] = {
          pos = pos,
          end_pos = pos,
        }
      end
      return res
    end,
```

BTW, Is that custom action really necessary?

Anyway yeah I do think this is really not worth it just for jumping. Its super easy to run out of labels (26*2 only) too

**Reply by folke** [reply]
Custom action is indeed not needed. Also see my other comment, which implements line mode in a much easier way. Unless I'm missing something?

**Reply by IndianBoy42** [reply]
Mine is jumping vertically (see how it uses win_get_cursor) rather than jumping to the beginning of the line. Some people like it more since I remember hop and easy-motion had this as a command

**Reply by max397574** [reply]
this can just be done with `<count>jk` (I see that there is a difference though) whereas there is no builtin way to easily also get to the beginning of the lines without just `0`

**Comment by folke** [comment]
Something like this?

```lua
require("flash").jump({
  search = { mode = "search" },
  pattern = "^"
})

**Reply by max397574** [reply]
when I use this this the labels of the lines are at the end and not the beginning which makes it way less useful imo

**Reply by folke** [reply]
Good point. Use this instead:

```lua
require("flash").jump({
  search = { mode = "search" },
  highlight = { label = { after = { 0, 0 } } },
  pattern = "^",
})
```

![image](https://github.com/folke/flash.nvim/assets/292349/01b5baf2-6535-4ff6-b53a-8c7992ee674d)

**Reply by Insofan** [reply]
I have added sub code
`require("flash").jump({
  search = { mode = "search" },
  highlight = { label = { after = { 0, 0 } } },
  pattern = "^",
})`,but I don't know how to press it on keyboard

**Reply by max397574** [reply]
just add a mapping that calls this function
`:h vim.keymap.set()`
